### PR TITLE
[C# SDK] Add SayAsync extension method

### PR DESCRIPTION
**Pull request by ejadib**
Some things to provide feedback on:

- Location of the MessageOptions class
- Location of the unit tests for the SayAsync extension method

### CODE REVIEW

File: CSharp/Library/Microsoft.Bot.Builder/Dialogs/IDialogContext.cs [hunk 1]
```diff
@@ -109,6 +149,44 @@ public static partial class Extensions
...
+        /// <summary>
+        /// Post a message and optional SSML to be sent to the user, using previous messages to establish a conversation context.
+        /// </summary>
+        /// <remarks>
+        /// If the locale parameter is not set, locale of the incoming message will be used for reply.
+        /// </remarks>
+        /// <param name="botToUser">Communication channel to use.</param>
+        /// <param name="text">The message text.</param>
+        /// <param name="speak">The SSML markup for text to speech.</param>
+        /// <param name="options">The optioms for the message.</param>
```
Comment (willportnoy) [inline CSharp/Library/Microsoft.Bot.Builder/Dialogs/IDialogContext.cs]: "options" spelling

File: CSharp/Library/Microsoft.Bot.Builder/Dialogs/IDialogContext.cs [hunk 2]
```diff
@@ -109,6 +149,44 @@ public static partial class Extensions
...
+
+            message.Text = text;
+            message.Speak = speak;
+
+            if (!string.IsNullOrEmpty(locale))
+            {
+                message.Locale = locale;
+            }
+
+            if (options != null)
```
Comment (willportnoy) [inline CSharp/Library/Microsoft.Bot.Builder/Dialogs/IDialogContext.cs]: is it it better to package these into a class or to use keyword named arguments?

File: CSharp/Library/Microsoft.Bot.Builder/Dialogs/IDialogContext.cs (hunk 1 above)
Comment (ejadib) [inline CSharp/Library/Microsoft.Bot.Builder/Dialogs/IDialogContext.cs]: fixed

File: CSharp/Library/Microsoft.Bot.Builder/Dialogs/IDialogContext.cs (hunk 2 above)
Comment (ejadib) [inline CSharp/Library/Microsoft.Bot.Builder/Dialogs/IDialogContext.cs]: Are you referring to the MessageOptions class? I’m fine either way, I added a class so it’s aligned to what Steve did. He has an interface [IMessageOptions](https://github.com/Microsoft/BotBuilder/blob/master/Node/core/lib/botbuilder.d.ts#L106-L122). We can change it to keyword named arguments if that makes more sense. LMK

**Pull request merged**